### PR TITLE
Implement/derive Canon for all 'groups' types

### LINKED ISSUE #42: Derive/implement Canon for structs that fall under "groups" feature
This is needed for https://github.com/dusk-network/bls12_381-sign/issues/7

### PR DESCRIPTION
**Pull request by julesdesmit**
Fixes #42

### CODE REVIEW

**Review by vlopes11 — APPROVED** [review]
LGTM

File: src/g2.rs [hunk 1]
```diff
@@ -26,6 +30,27 @@ pub struct G2Affine {
     infinity: Choice,
 }
 
+#[cfg(feature = "canon")]
+impl<S: Store> Canon<S> for G2Affine {
+    fn write(&self, sink: &mut impl Sink<S>) -> Result<(), S::Error> {
+        self.x.write(sink)?;
+        self.y.write(sink)?;
```
Comment (CPerezz) [inline src/g2.rs:37]: Why not compressing the point as bytes? That will reduce the size and grant us a check when we use read.

Anyway, LGTM.

File: src/g1.rs [hunk 2]
```diff
@@ -24,6 +28,27 @@ pub struct G1Affine {
     infinity: Choice,
 }
 
+#[cfg(feature = "canon")]
+impl<S: Store> Canon<S> for G1Affine {
+    fn write(&self, sink: &mut impl Sink<S>) -> Result<(), S::Error> {
+        self.x.write(sink)?;
+        self.y.write(sink)?;
```
Comment (CPerezz) [inline src/g1.rs:35]: ditto

**Review by CPerezz — APPROVED** [review]
Just a few suggestions.

But LGTM anyway.

**Comment by CPerezz** [comment]
@julesdesmit should this be merged??

**Comment by julesdesmit** [comment]
@CPerezz yes, I was just waiting on @ZER0's review. But if you deem it correct, feel free to merge.

**Comment by CPerezz** [comment]
We usually say that with 1 approval is enough to merge. So feel free to do so @julesdesmit . Since this PR is required for the new release of jubjub.

**Pull request merged**